src/main/java/vuewee/TaskListUI.java

```
@@ -0,0 +1,129 @@
+            }
+            if (isDone) {
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+            System.out.println("  " + this.taskList.get(taskNumber).toString());
+        } catch (IndexOutOfBoundsException e) {
+            throw new IllegalCommandException("Invalid task number. There are " + this.taskList.size() + " "
+                    + this.taskWord() + " in your list.");
+        }
+    }
+
+    public void run() {
+        this.taskList = storage.readTasks();
+        if (this.taskList.size() > 0) {
+            System.out.println("Loaded " + this.taskList.size() + " " + this.taskWord() + " into your task list.");
+        }
+
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Vuewee\nWhat can I do for you?");
+        System.out.println("____________________________________________________________");
+
+        // Echo input from user until user types "bye"
+        try {
+            while (true) {
+                String input = scanner.nextLine();
+                try {
+                    System.out.println("____________________________________________________________");
+
+                    CommandParser parser = new CommandParser(input);
+                    CommandType commandType = parser.getCommandType();
+                    Command command = commandType.createCommand();
+                    command.execute(this, taskList, parser);
+
+                } catch (IndexOutOfBoundsException | IllegalCommandException e) {
+                    System.out.println(e.getMessage());
+                }
+                System.out.println("____________________________________________________________");
+                storage.storeTasks(this.taskList);
+            }
+        } catch (EndProgramException e) {
```

Empty catch block present here, perhaps there can be a way to handle these exceptions such as print the issue out for the user?